optuna/study/_multi_objective.py

```diff
@@ -127,17 +126,18 @@ def _is_pareto_front_nd(unique_lexsorted_loss_values: np.ndarray) -> np.ndarray:
     loss_values = unique_lexsorted_loss_values[:, 1:]
     n_trials = loss_values.shape[0]
     on_front = np.zeros(n_trials, dtype=bool)
-    nondominated_indices: np.ndarray[tuple[int, ...], np.dtype[np.signedinteger[Any]]] = np.arange(
-        n_trials
-    )
+    nondominated_indices = np.arange(n_trials)
     while len(loss_values):
         # The following judges `np.any(loss_values[i] < loss_values[0])` for each `i`.
         nondominated_and_not_top = np.any(loss_values < loss_values[0], axis=1)
         # NOTE: trials[j] cannot dominate trials[i] for i < j because of lexsort.
         # Therefore, nondominated_indices[0] is always non-dominated.
         on_front[nondominated_indices[0]] = True
         loss_values = loss_values[nondominated_and_not_top]
-        nondominated_indices = nondominated_indices[nondominated_and_not_top]
+        # NOTE(nabenabe0928): Ignore typing for a temporal solution to NumPy v2.2.0 weird typing.
+        nondominated_indices = nondominated_indices[
+            nondominated_and_not_top
+        ]  # type: ignore[assignment]
```

Comment:
I could not find any good ways to tell NumPy that the return of this vectorization is gonna be a 1d integer array, so I ignore the typing here for now.

Reply:
I found this option, so I replaced the code.
- https://github.com/numpy/numpy/issues/27957#issuecomment-2531536212